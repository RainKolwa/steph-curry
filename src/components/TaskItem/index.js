import React from 'react'
import PropTypes from 'prop-types'
import './style.styl'

const TaskItem = ({ task }) => {
  const { type, title, completed, total } = task
  return (
    <div className="task-item">
      <span>{type}：</span>
      <span>{title}</span>
      <div>
        {completed}/{total}
      </div>
    </div>
  )
}

TaskItem.propTypes = {
  task: PropTypes.shape({
    type: PropTypes.oneOf(['预习', '作业', '上课']),
    title: PropTypes.string.isRequired,
    completed: PropTypes.number.isRequired,
    total: PropTypes.number.isRequired,
  }).isRequired,
}

export default TaskItem
