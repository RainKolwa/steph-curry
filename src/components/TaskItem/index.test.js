import React from 'react'
import renderer from 'react-test-renderer'
import TaskItem from './index'

it('renders correctly', () => {
  const task = {
    type: '上课',
    title: '人们如何帮助社区？',
    completed: 2,
    total: 10,
  }
  const userhead = renderer.create(<TaskItem task={task} />).toJSON()
  expect(userhead).toMatchSnapshot()
})
