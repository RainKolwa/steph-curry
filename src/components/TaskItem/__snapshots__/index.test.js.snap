exports[`test renders correctly 1`] = `
<div
  className="task-item">
  <span>
    上课
    ：
  </span>
  <span>
    人们如何帮助社区？
  </span>
  <div>
    2
    /
    10
  </div>
</div>
`;
